src/dashboard/src/components/helpers.py

```diff
@@ -300,32 +303,81 @@ def processing_config_path():
     )
 
 
+def _prepare_stream_response(
+    payload, content_type, content_disposition, preview_file=False
+):
+    """Prepare the streaming response to return to the caller."""
+    if preview_file:
+        content_disposition = "inline"
+    response = StreamingHttpResponse(payload)
+    response["Content-Type"] = content_type
+    response["Content-Disposition"] = content_disposition
+    return response
+
+
+def stream_mets_from_storage_service(
+    transfer_name, sip_uuid, error_message="Unexpected error: {}"
+):
+    """Enable the streaming of an individual AIP METS file from the Storage
+    Service.
+    """
+    absolute_transfer_name = "{}-{}".format(transfer_name, sip_uuid)
+    mets_name = "METS.{}.xml".format(sip_uuid)
+    mets_path = "{}/data/{}".format(absolute_transfer_name, mets_name)
+    mets_tmp_dir = tempfile.mkdtemp()
+    mets_tmp_file = os.path.join(mets_tmp_dir, mets_name)
+    # We can't get a lot of debug information from AMClient yet, so we try to
+    # download and then open, returning an error if the file can't be accessed.
+    try:
+        AMClient(
+            ss_api_key=get_setting("storage_service_apikey", None),
+            ss_user_name=get_setting("storage_service_user", "test"),
+            ss_url=get_setting("storage_service_url", None).rstrip("/"),
+            package_uuid=sip_uuid,
+            relative_path=mets_path,
+            saveas_filename=mets_tmp_file,
+        ).extract_file()
+    except requests.exceptions.ConnectionError as err:
+        response = {"success": False, "message": error_message.format(err)}
+        return json_response(response, status_code=503)
+    mets = ""
+    err_msg = None
+    try:
+        with open(mets_tmp_file, "r") as mets_file:
+            mets = mets_file.read()
+        os.unlink(mets_tmp_file)
+    except IOError as err:
+        err_msg = err
+    if not mets:
+        response = {"success": False, "message": error_message.format(err_msg)}
+        return json_response(response, status_code=404)
```

Comment:
How hard would it be to add a `stream_file()` function to AMClient? Temporarily loading the file on the filesystem seems a bit to much to only re-stream it later.

Reply:
@replaceafill asked about this also, but I don't believe it to be in scope. I noted to him that this is an incremental step in enhancing both the behavior of Archivematica and AMClient and so it is something I believe we can react to if there proves to be a problem with this mechanism. Although admittedly, AMClient weaknesses have previously been noted too, it's just the first time we're able to find a good reason to integrate it :slightly_smiling_face:

Reply:
I understand this may not be in scope and I'm sorry I'm late to the code review. Nevertheless, from a SAD perspective, this is far from ideal. In general, the current process would be:

- The user clicks the view METS file button.
- The file is extracted and downloaded from the SS to the filesystem. Luckily, the file is not fully loaded on memory [on that process](https://github.com/artefactual-labs/amclient/blob/ee66a2e22a76e89ad1f9c47012eb2336c6a78a76/amclient/amclient.py#L599-L630).
- However, the file is fully loaded in memory in here before deleting it from the filesystem.
- A response is finally sent to the user, streaming the METS file content.

Now that I see the `extract_file` function, I think you could easily achieve the stream behavior. It doesn't have to be a new function, it could be an optional parameter on that function, `return_stream` for example. When it's set to `true` the response is returned directly [here](https://github.com/artefactual-labs/amclient/blob/ee66a2e22a76e89ad1f9c47012eb2336c6a78a76/amclient/amclient.py#L608) before the file is loaded on the filesystem.

Reply:
Thanks @jraddaoui I hope you appreciate the technical aspects here are not lost on me (silly bugs aside). Rather my hesitation (as always is) discussing with both you and @replaceafill were around the other organizational context a new feature finds itself wrapped within within. Thank you for taking the time to review this on your weekend. 

From a CR perspective, I appreciate you taking the time to propose something based on the code that was in AMClient and spotting the short cut to adding streaming. Where there is room to provide mentorship in our work then that's definitely an approach that works for me, and I'm sure others. I also try and do that work up-front where it can help devs.

For others reading, the PR is [here](https://github.com/artefactual-labs/amclient/pull/13). You'll see a new helper function that enables streaming to the command-line part of AMClient too, plus the changes to make streaming possible. And I've added to ability to `stream` -- ostensibly our streaming flag to be used by the original `extract_file` we touch upon here.

Reply:
Thank you @ross-spencer! I really appreciate the effort.

Of course, I also appreciate the technical aspects of this changes and I like the idea to introduce `amclient`. I hope that, with a few enhancements, it could replace the other helpers across the code, but without the streaming capability I wouldn't have considered it as a good option for this feature.